### ISSUE TITLE
Ticket from Imap mail not being created

### ISSUE THREAD

**Issue by sukhbirgs** [uvdesk/core-framework#171]
Using command line to get tickets from imap account, connection is made and 2 emails processed but no ticket created.

**/usr/local/php72/bin/php bin/console uvdesk:refresh-mailbox sales@help.domain.com**

Got following confirmation but no ticket was created

```
1. Processing uvdesk mailbox configuration.

 2. Opening imap stream... 

 3. IMAP stream opened.

 4. Fetching Email collection since 10 July 2019.

 5. Total fetched email -> 2

 6. Starting to convert Emails into Tickets -> 
=============================================
=============================================


  Converting email 1 out of 2.

  Converting email 2 out of 2.

 Mailbox refreshed successfully !!!
```

**Comment by papnoisanjeev** [comment]
Please share screen shot of **uvdesk.yaml,  uvdesk_mailbox.yaml**.

**Comment by sukhbirgs** [comment]
Here you go, thanks

![uvdesk](https://user-images.githubusercontent.com/74910/61140137-1c5ee780-a488-11e9-9af9-b9e702818206.JPG)
![uvdesk_mailbox](https://user-images.githubusercontent.com/74910/61140138-1cf77e00-a488-11e9-9841-4c3b1989cdbc.JPG)

**Comment by sukhbirgs** [comment]
I ran console command with debug enabled and got this error

`
 1. Processing uvdesk mailbox configuration.

 2. Opening imap stream... 

 3. IMAP stream opened.

 4. Fetching Email collection since 11 July 2019.

 5. Total fetched email -> 1

 6. Starting to convert Emails into Tickets -> 
=============================================
=============================================


  Converting email 1 out of 1.

Mailbox refreshed successfully !!!

**09:58:13 INFO      [php] User Deprecated: Calling the "Symfony\Component\EventDispatcher\EventDispatcherInterface::dispatch()" method with the event name as first argument is deprecated since Symfony 4.3, pass it second and provide the event object first instead.
[
  "exception" => ErrorException {
    #message: "User Deprecated: Calling the "Symfony\Component\EventDispatcher\EventDispatcherInterface::dispatch()" method with the event name as first argument is deprecated since Symfony 4.3, pass it second and provide the event object first instead."
    #code: 0
    #file: "/home/mohelp/domains/help.mydomain.com/public_html/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php"
    #line: 148
    #severity: E_USER_DEPRECATED
    trace: {
      /home/mohelp/domains/help.mydomain.com/public_html/vendor/symfony/event-dispatcher/Debug/TraceableEventDispatcher.php:148 { â€¦}
      /home/mohelp/domains/help.mydomain.com/public_html/vendor/symfony/console/Application.php:1014 { â€¦}
      /home/mohelp/domains/help.mydomain.com/public_html/vendor/symfony/framework-bundle/Console/Application.php:86 { â€¦}
      /home/mohelp/domains/help.mydomain.com/public_html/vendor/symfony/console/Application.php:255 { â€¦}
      /home/mohelp/domains/help.mydomain.com/public_html/vendor/symfony/framework-bundle/Console/Application.php:74 { â€¦}
      /home/mohelp/domains/help.mydomain.com/public_html/vendor/symfony/console/Application.php:148 { â€¦}
      /home/mohelp/domains/help.mydomain.com/public_html/bin/console:42 {
        â€º $application = new Application($kernel);
        â€º $application->run($input);
        â€º 
      }
    }
  }
]**
`

**Comment by papnoisanjeev** [comment]
**vendor >> uvdesk >> mailbox-component >> console >> RefreshMailboxCommand.php**  open this file and (dump step by step your inputs and outputs) check site url and other details  and try to debug step by step why mails are not converting into tickets.

**Comment by sukhbirgs** [comment]
Debugged said file and i found issue in $curlResponse in pushMessage function

$requestUrl = http://help.domain.com/public/mailbox/listener

curl error: 
http://help.domain.com/public/mailbox/listener
The requested URL returned error: 403 Forbidden
[config->packages->uvdesk.yaml , $site_url = help.domain.com/public/]

Error printout
`<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>403 Forbidden</title>
</head><body>
<h1>Forbidden</h1>
<p>You don't have permission to access /public/mailbox/listener
on this server.<br />
</p>
</body></html>`

**Comment by papnoisanjeev** [comment]
Hello sukhbirgs,

Please remove **/** after public from site url **$site_url = help.domain.com/public/**.
and then run mailbox refresh command.

**Comment by sukhbirgs** [comment]
My mistake in

> Hello sukhbirgs,
> 
> Please remove **/** after public from site url **$site_url = help.domain.com/public/**.
> and then run mailbox refresh command.

Sorry my mistake, $site_url is without trailing slash 
 **$site_url = help.domain.com/public**
Screenshot of my uvdesk.yml is posted above. Thanks

**Comment by papnoisanjeev** [comment]
Hello Sukhbirgs,

The problem may be due to you are not able to access your public folder please check your .htaccess file and try to resolve.
check this:
https://stackoverflow.com/questions/18272557/laravel-forbidden-you-dont-have-permission-to-access-on-this-server/38490462

**Comment by vaishaliwebkul** [comment]
Hello @sukhbirgs 

 We haven't found any reply from your end yet so we are going to close this issue.
Please update if stuck anywhere :)

**Comment by arizonal2014** [comment]
Hello there!

I am having the same issues. Is there any solution to this?

Regards
Harrison

**Comment by papnoisanjeev** [comment]
@arizonal2014 

Have you checked your **site_url**.
Are you able to see conversion of email same like [here](https://github.com/uvdesk/core-framework/issues/171#issuecomment-510941839).

**Comment by arizonal2014** [comment]
@papnoisanjeev Yes. Yesterday I did spin up centos 7 and 8. Ubuntu 16, 18, and 20 LTS, and carefully install it, unfortunately, it does not work. I have also read all the tickets about this topic. I suggest your team fix it.

**Comment by papnoisanjeev** [comment]
@arizonal2014 

Please share screenshots about errors you are getting if any so that we can check from our end.
It is not a general issue which everyone is facing so without knowing all steps and error screenshot it is not possible to guess what actually the issue is.

**Comment by softwareforenterprise** [comment]
Hi there,

I'm facing a similar issue where the error is showing Forbidden.

Here's the screenshot error when I dump($curlResponse);
![curl response](https://user-images.githubusercontent.com/88500383/131829604-27ba3450-32d4-47e6-a197-c4315401e8e3.png)

Can you provide some other ideas for me to check?

Would like to get the mailbox to begin creating tickets from emails.

Thanks,
Lee

**Comment by vipin-shrivastava** [comment]
@softwareforenterprise 

Are you not able to convert your email into  ticket ?

Which version of uvdesk are you using ?

**Comment by vipin-shrivastava** [comment]
@softwareforenterprise 

Please share screen shot of uvdesk.yaml, uvdesk_mailbox.yaml.

**Comment by softwareforenterprise** [comment]
> @softwareforenterprise
> 
> Are you not able to convert your email into ticket ?
> 
> Which version of uvdesk are you using ?

Thanks for your reply Vipin. I'm using version  V -1.0.14

**Comment by vipin-shrivastava** [comment]
@softwareforenterprise 
Please share screen shot of uvdesk.yaml, uvdesk_mailbox.yaml.

**Comment by softwareforenterprise** [comment]
> @softwareforenterprise
> Please share screen shot of uvdesk.yaml, uvdesk_mailbox.yaml.

@vipin-shrivastava 

Here's screen shot of uvdesk.yaml, uvdesk_mailbox.yaml.
![uvdesk_mailbox](https://user-images.githubusercontent.com/88500383/131842652-bcb244ad-05ec-4987-a367-ba438c0a1267.png)
![uvdesk](https://user-images.githubusercontent.com/88500383/131842667-42a37b2d-e146-4ac8-ba41-a5324ceddb25.png)

**Comment by vipin-shrivastava** [comment]
@softwareforenterprise 
I think your site_url is not correct, 
Example : 
`site_url: 'localhost/helpdesk/public'`

![Screenshot_21](https://user-images.githubusercontent.com/58874186/131947417-c5901e0f-756a-4b2f-97e4-88f0f516fd62.png)

I think you are missing the project name in **site_url.**. check it again If your issue still exists please let me know

**Comment by softwareforenterprise** [comment]
@vipin-shrivastava 
I've tried to include the project name folder into the site_url but now it's show 404 error instead of the initial Forbidden Access error. The ticket is also not created.
![404 error](https://user-images.githubusercontent.com/88500383/131952711-2e0085c5-cb2e-4785-8cfe-0b89b48a75a2.png)

When accessing the Request_url via a browser, it returns a 200 code. However when I run this command from SSH php bin/console uvdesk:refresh-mailbox correct_email_address_here it returns Forbidden Access.

Do you know if there's another way to execute the Request_url?

**Comment by vipin-shrivastava** [comment]
@softwareforenterprise 
Do one thing save your **Email Settings** in the Settings section.
![Screenshot_22](https://user-images.githubusercontent.com/58874186/131954113-651b2c8d-39a3-4992-8a53-beb683c13db2.png)
Then save your Email settings. It will automatically update your site_url.
![Screenshot_23](https://user-images.githubusercontent.com/58874186/131954130-77358dca-828b-4d15-822b-cb51c99f1fc8.png)

After update, the email setting then try again with Mailbox:refresh command 
Hope your issue will be resolved if not please let us know

**Comment by vipin-shrivastava** [comment]
@softwareforenterprise 

Please share your helpdesk URL, So that we can check.

**Comment by softwareforenterprise** [comment]
@vipin-shrivastava 

I've tried the steps but issue still not resolved.

May I know what's the best way for me to contact you directly rather than posting the URL publicly here?

Thank you for your help!

**Comment by vipin-shrivastava** [comment]
@softwareforenterprise 

Mail your issue to Support@webkul.com. Our support team members will help you.

**Comment by vipin-shrivastava** [comment]
@softwareforenterprise 

Mail your issue to Support@uvdesk.com. Our support team members will help you.

**Comment by softwareforenterprise** [comment]
Hi! For anyone facing a similar issue, here's what we did to solve the issue. Thanks again to Vipin for your kind assistance and guidance.

The issue I faced:
1. Mailbox settings and cron are set up correctly. However, emails from mailbox did not convert to tickets.
2. My environment is shared hosting, installed via softaculous in a subdomain, modsecurity enabled by default.
3. After putting dump($requestUrl.': '.$curlResponse);  inside the code RefreshMailboxCommand.php and running refresh-mailbox via SSH, the returned error was 403 Forbidden, access to this resource on the server is denied!
4. UVDesk Support team recommended checking with hosting regarding any security enabled which blocks the URL.
5. I contacted my web hosting support and explained the situation. 
6. All they did was whitelist the URL because modsecurity kicked in to trigger the block. Once whitelisted, the tickets were created just fine.

Hope this helps anyone facing a similar issue.
